### PR TITLE
[Draft] Add pokemonImage to notify Toast, add icons to all pokemon notifications

### PR DESCRIPTION
**Pull request by YetiSpaghetti**
Addresses Pokemon part of https://github.com/pokeclicker/pokeclicker/issues/3904

Adds Pokemon icon to all standard notifications involving Pokemon that I could find

- Caught
- Hatched
- Shiny Encounter (using shiny sprite)
- Shiny Caught (using shiny sprite)

Note that this uses the basic sprite assets. While having a party icon assets would ultimately be better in term of uniform pokemon scale, it would require adding thousands of additional assets _and_ handling missing icons. FWIW these sprites are already used in a toast for Pokemon Encounters from Dungeon Chests

```ts
message = `Encountered ${GameHelper.anOrA(input)} <img src="${image}" height="40px"/> ${GameConstants.humanifyString(input)} in a dungeon chest.`;
```

I decided to add a new `pokemonImage` parameter and `pokemonIcon` css class to the toast notifier for Pokemon. This is due to the default icon size leading to small pokemon sprites being extremely difficult to see. 

This class makes all PokemonIcons bigger while maintaining the overall toast dimensions of a normal icon (text alignment, toast height, etc)

In the below screenshots, Pidgeotto/Fearow are slightly smaller than the largest pokemon sprite, and Bulbasaur/Rattata are some of the smallest. For reference, the berries are the standard toast icon sizing. Everything is easy to perceive, and has consistent margins with the text, and the pokemonIcons do not change the standard height of toasts.

| Icon class | PokemonIcon class |
| --- | --- | 
| image | image |

### CODE REVIEW

File: src/modules/notifications/Notifier.ts [hunk 1]
```diff
@@ -23,6 +24,7 @@ export default class Notifier {
         sound?: Sound;
         setting?: NotificationSetting;
         image?: string;
+        pokemonImage?: string;
```
[the remark this answers is not in the document]
Comment (YetiSpaghetti) [reply src/modules/notifications/Notifier.ts:27]: I also considered just embedding the icons directly in the notification message, but this feels very brittle, and I think it's a common enough use case to warrant it's own parameter/class on the Notifier

**Comment by Leafsw0rd** [comment]
Would it look better to use Shuffle sprites? Those have significantly more uniformity in their style and size. 
image

**Comment by DraKay** [comment]
And don't include Gen 8 or alt forms

**Comment by YetiSpaghetti** [comment]
Yeah ultimately I think party sprites is the "optimal" solution from an aesthetic PoV, but that would require adding a ton more assets to the bundle. I'll let the more experienced contributors decide if that tradeoff is worth it (if this suggestion even gets approved, this was just an exploratory PR)

**Pull request merged**